### ISSUE TITLE
Make generation of auto-completions cancelable

### ISSUE THREAD

**Issue by szschaler** [gts-morpher/gts_morpher#49]
There can be a lot of auto-completions to generate for some partial morphism specifications. This can make the process of generating auto-completions long and can end up blocking up Eclipse completely.

We already use a progress monitor, but currently aren't checking for cancellation requests (and, in particular, we're not using it at all in the code that actually calculates all completions). It would also be good to change the actual calculation code to something that doesn't require calculating all completions before we can start exporting them, but rather something that provides an iterator from which we can extract as few or as many completions as we are interested in (though this may be better as a separate issue).
